binding/python/test_koala.py

```diff
@@ -0,0 +1,103 @@
+#
+#    Copyright 2023 Picovoice Inc.
+#
+#    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+#    file accompanying this source.
+#
+#    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+#    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#    specific language governing permissions and limitations under the License.
+#
+import argparse
+import os
+import struct
+import sys
+import unittest
+import wave
+from typing import Optional, Sequence
+
+import numpy as np
+
+from _koala import Koala
+from _util import default_library_path, default_model_path
+
+
+class KoalaTestCase(unittest.TestCase):
+    ACCESS_KEY: str
+    AUDIO_PATH = os.path.join(os.path.dirname(__file__), '../../resources/audio_samples/test.wav')
+
+    pcm: Sequence[int]
+    koala: Koala
+
+    @classmethod
+    def setUpClass(cls):
+        with wave.open(os.path.join(os.path.dirname(__file__), '../../resources/audio_samples/test.wav'), 'rb') as f:
+            buffer = f.readframes(f.getnframes())
+            cls.pcm = struct.unpack('%dh' % f.getnframes(), buffer)
+
+        cls.koala = Koala(
+            access_key=cls.ACCESS_KEY,
+            model_path=default_model_path('../..'),
+            library_path=default_library_path('../..'))
+
+    def test_frame_length(self) -> None:
+        self.assertGreater(self.koala.frame_length, 0)
+
+    def test_delay_sample(self) -> None:
+        self.assertGreaterEqual(self.koala.delay_sample, 0)
+
+    @staticmethod
+    def _pcm_root_mean_square(pcm: Sequence[int]) -> float:
+        return np.sqrt(np.mean(np.square(np.array(pcm) / (2 ** 15))))
+
+    def _run_test(
+            self,
+            input_pcm: Sequence[int],
+            reference_pcm: Optional[Sequence[int]] = None,
+            tolerance: float = 0.01) -> None:
+        num_samples = len(input_pcm)
+        frame_length = self.koala.frame_length
+        delay_sample = self.koala.delay_sample
+
+        self.koala.reset()
```

Comment:
we prob need a separate test for `reset` to make sure it actually wipes all the states